mlflow/data/evaluation_dataset.py

```diff
@@ -277,7 +277,12 @@ def __init__(
             # add checking `'pyspark' in sys.modules` to avoid importing pyspark when user
             # run code not related to pyspark.
             if "pyspark" in sys.modules:
-                from pyspark.sql import DataFrame as SparkDataFrame
+                from mlflow.utils.spark_utils import is_spark_connect_mode
+
+                if is_spark_connect_mode():
+                    from pyspark.sql.connect.dataframe import DataFrame as SparkDataFrame
+                else:
+                    from pyspark.sql import DataFrame as SparkDataFrame
```

Comment:
any chance that the normal dataframe even in the connect?

Reply:
are there more places where `pyspark.sql.connect.dataframe.DataFrame` is missing?

Reply:
> any chance that the normal dataframe even in the connect?

I think no.

Reply:
confirmed with @HyukjinKwon , spark dataframe is always `pyspark.sql.connect.dataframe.DataFrame` type.
so we can only set spark df type to `pyspark.sql.connect.dataframe.DataFrame`  , otherwise `EvaluationDataset._spark_df_type` need to be updated too, which is unnecessary